InvenTree/web/urls.py

```diff
@@ -0,0 +1,26 @@
+"""URLs for web app."""
+from django.conf import settings
+from django.shortcuts import redirect
+from django.urls import path, re_path
+from django.views.decorators.csrf import ensure_csrf_cookie
+from django.views.generic import TemplateView
+
+
+class RedirectAssetView(TemplateView):
+    """View to redirect to static asset."""
+
+    def get(self, request, *args, **kwargs):
+        """Redirect to static asset."""
+        return redirect(
+            f"{settings.STATIC_URL}web/assets/{kwargs['path']}", permanent=True
+        )
+
+
+spa_view = ensure_csrf_cookie(TemplateView.as_view(template_name="web/index.html"))
+
+
+urlpatterns = [
+    path("assets/<path:path>", RedirectAssetView.as_view()),
+    re_path(r"^(?P<path>.*)/$", spa_view),
+    path("", spa_view),
```

Comment:
What is the advantage about passing this as a template instead of using only static files for the spa? Doesn't this also have an impact about performance?

Reply:
CSRF protection and session management mostly. The data is also only loaded once/redirected so the proxy will only load full loads from the backend, everything else is cached/served from static.